pkg/webhook/webhook.go

```diff
@@ -399,22 +400,30 @@ func patchEmptyResources(patch []jsonPatchOperation, containerIndex uint, key st
 	return patch
 }
 
-func addVolDownwardAPI(patch []jsonPatchOperation, hugepageResourceList []hugepageResourceData) []jsonPatchOperation {
-	labels := corev1.ObjectFieldSelector{
-		FieldPath: "metadata.labels",
-	}
-	dAPILabels := corev1.DownwardAPIVolumeFile{
-		Path:     types.LabelsPath,
-		FieldRef: &labels,
-	}
-	annotations := corev1.ObjectFieldSelector{
-		FieldPath: "metadata.annotations",
+func addVolDownwardAPI(patch []jsonPatchOperation, hugepageResourceList []hugepageResourceData, pod *corev1.Pod) []jsonPatchOperation {
+	dAPIItems := []corev1.DownwardAPIVolumeFile{}
+
+	if pod.Labels != nil && len(pod.Labels) > 0 {
```

Comment:
@Billy99 Will this affect app-netutils from parsing the mounted file?  For example, if there is no labels, the mounted podNetInfo file will not contain labels key.

Reply:
I guess if this matters, then we would change app-netutils to accommadate.

Reply:
@zshi-redhat Sorry, missed your original comment. No, app-netutils doesn't currently look at the labels. And in general if something is not there (like annotations), it keeps going and does the best it can.

Reply:
Thanks @Billy99 for the confirmation!